components/match2/wikis/clashroyale/match_group_input_custom.lua

```diff
@@ -0,0 +1,357 @@
+end
+
+---@param match table
+---@param opponents table[]
+---@return table[]
+function MatchFunctions.extractMaps(match, opponents)
+	local maps = {}
+	local subGroup = 0
+	for mapKey, mapInput, mapIndex in Table.iter.pairsByPrefix(match, 'map', {requireIndex = true}) do
+		local map
+		map, subGroup = MapFunctions.readMap(mapInput, mapIndex, subGroup, #opponents)
+
+		map.participants = MapFunctions.getParticipants(mapInput, opponents)
+		map.extradata = MapFunctions.getExtraData(mapInput, map.participants)
+
+		map.vod = Logic.emptyOr(mapInput.vod, match['vodgame' .. mapIndex])
+
+		table.insert(maps, map)
+		match[mapKey] = nil
+	end
+
+	return maps
+end
+
+---@param maps table[]
+---@param opponents table[]
+---@return fun(opponentIndex: integer): integer?
+function MatchFunctions.calculateMatchScore(maps, opponents)
+	return function(opponentIndex)
+		local calculatedScore = MatchGroupInputUtil.computeMatchScoreFromMapWinners(maps, opponentIndex)
+		if not calculatedScore then return end
+		local opponent = opponents[opponentIndex]
+		return calculatedScore + (opponent.extradata.advantage or 0) - (opponent.extradata.penalty or 0)
+	end
+end
+
+---@param bestofInput string|integer?
+---@return integer?
+function MatchFunctions.getBestOf(bestofInput)
+	local bestof = tonumber(bestofInput) or tonumber(Variables.varDefault('match_bestof'))
+
+	if bestof then
+		Variables.varDefine('match_bestof', bestof)
+	end
+
+	return bestof
+end
+
+---@param match table
+---@param numberOfGames integer
+---@return table
+function MatchFunctions.getExtraData(match, numberOfGames)
+	local extradata = {
+		t1bans = MatchFunctions.readBans(match.t1bans),
+		t2bans = MatchFunctions.readBans(match.t2bans),
+		mvp = MatchGroupInputUtil.readMvp(match),
+	}
+
+	local prefix = 'subgroup%d+'
+	Table.mergeInto(extradata, Table.filterByKey(match, function(key) return key:match(prefix .. 'header') end))
+	Table.mergeInto(extradata, Table.filterByKey(match, function(key) return key:match(prefix .. 'iskoth') end))
+
+	local t1Bans = Table.filterByKey(match, function(key) return key:match(prefix .. 't1bans') end)
+	Table.mergeInto(extradata, Table.mapValues(t1Bans, MatchFunctions.readBans))
+
+	local t2Bans = Table.filterByKey(match, function(key) return key:match(prefix .. 't2bans') end)
+	Table.mergeInto(extradata, Table.mapValues(t2Bans, MatchFunctions.readBans))
+
+	return extradata
+end
+
+---@param bansInput string
+---@return table
+function MatchFunctions.readBans(bansInput)
+	local bans = CustomMatchGroupInput._readCards(bansInput)
+
+	return Logic.nilIfEmpty(bans)
+end
+
+---@param mapInput table
+---@param mapIndex integer
+---@param subGroup integer
+---@param opponentCount integer
+---@return table
+---@return integer
+function MapFunctions.readMap(mapInput, mapIndex, subGroup, opponentCount)
+	subGroup = tonumber(mapInput.subgroup) or (subGroup + 1)
+
+	local map = {
+		map = 'Set ' .. mapIndex, -- CR has no map names, use generic one instead
```

Comment:
```suggestion
```

Show this in the display instead

Reply:
display doesn't access it anyways, will just kick without additional changes needed